packages/manager/apps/vrack-services/src/api/order/hook.ts

```diff
@@ -0,0 +1,97 @@
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import {
+  getOrderStatus,
+  getOrderDetails,
+  getOrderDetailsList,
+  getOrderList,
+} from './services';
+
+export const toDetailedOrder = async (orderId: number) => {
+  const orderProductList = await getOrderDetailsList(orderId);
+  const detailsResponse = await Promise.all(
+    orderProductList.data.map((detailId) =>
+      getOrderDetails({ orderId, detailId }),
+    ),
+  );
+  return {
+    orderId,
+    details: detailsResponse.map((detail) => detail.data),
+  };
+};
+
+export const getDeliveringOrderList = (
+  orderDetailDescription: string,
+) => async () => {
+  const todayMinus24Hours = new Date();
+  todayMinus24Hours.setTime(Date.now() - 24 * 3600 * 1000);
+  const orderList = await getOrderList({ dateFrom: todayMinus24Hours });
+
+  const orderIdsWithStatus = await Promise.all(
+    orderList.data.map(async (orderId) => {
+      const status = await getOrderStatus(orderId);
+      return { orderId, status: status.data };
+    }),
+  );
+  const deliveringOrderList = orderIdsWithStatus
+    .filter(({ status }) => status === 'delivering')
+    .map(({ orderId }) => orderId);
+
+  const detailedOrderList = await Promise.all(
+    deliveringOrderList.map(toDetailedOrder),
+  );
+  return (
+    detailedOrderList
```

Comment:
Could you extract this to a dedicated `function` with appropriate name it's hard to understand what buisness logic you're applying here

Reply:
https://github.com/ovh/manager/pull/10330/files#diff-c3e6cef7ec25edbc0c400c8a033f3cf062b38dfbea2a5b84f6a5e22383e0de82

done here